File: go/060.go
func getPermutation(n int, k int) string {
    nums := make([]int, n)
    for i := 1; i <= n; i++ {
        nums[i-1] = i
    }
    var res string
    for len(nums) > 0 {
        divisor := fac(len(nums) - 1)
        idx := (k - 1) / divisor
        res += strconv.Itoa(nums[idx])
        k -= divisor * idx
        temp := nums[:idx]
        temp = append(temp, nums[idx+1:]...)
        nums = temp
    }
    return res
}

func fac(k int) int {
    ret := 1
    if k == 0 || k == 1 {
        return ret
    }
    for ; k > 1; k-- {
        ret *= k
    }
    return ret
}

File: cpp/060.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        vector<int> nums;
        for (int i = 1; i <= n; ++i)
            nums.push_back(i);
        string res;
        while (!nums.empty()) {
            int divisor = fac(nums.size() - 1);
            int idx = (k - 1) / divisor;
            res += nums[idx] + '0';
            k -= divisor * idx;
            nums.erase(nums.begin()+idx);
        }
        return res;
    }

    int fac(int n) {
        if (n == 0 || n == 1)
            return 1;
        int res = 1;
        for (int i = 2; i <= n; ++i)
            res *= i;
        return res;
    }
};
